### ISSUE TITLE
Parameter "oneOf" from openapi @Schema annotation is ignored during generation the OpenApi document

### ISSUE THREAD

**Issue by srnjak** [OpenLiberty/open-liberty#9709]
I found strange behavior in microprofile openapi schema generation using annotation.

I created a POJO with a property, which could hold an object or a list of objects. So I annotated it with `@Schema(oneOf = {Object.class, List.class})`. I expected that schema will be created using the `oneOf` keywoard.

It seems that `oneOf` property of `@Schema` annotation was just ignored and during generation of schema java type of property was used.

Schema class:
```
import org.eclipse.microprofile.openapi.annotations.media.Schema;
import java.util.List;

public class SampleSchema {

    @Schema(oneOf = {Object.class, List.class})
    public Object objectOrList;
}
```
Resource class
```
import org.eclipse.microprofile.openapi.annotations.media.Content;
import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

@Path("/test")
public class TestResource {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    @APIResponses({
            @APIResponse(responseCode = "200",
                    content = @Content(
                            schema = @Schema(implementation = SampleSchema.class)))
    })
    public Response get() {

        return Response.ok().build();
    }
}
```
Generated schema:
```
openapi: 3.0.0
info:
  title: Deployed APIs
  version: 1.0.0
servers:
- url: http://pingvinx:9080/media-archive
paths:
  /resources/test:
    get:
      operationId: get
      responses:
        200:
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/SampleSchema'
components:
  schemas:
    SampleSchema:
      type: object
      properties:
        objectOrList:
          type: object
```

Expected schema of the `SampleSchema` type would be:
```
    SampleSchema:
      type: object
      properties:
        objectOrList:
          oneOf:
            - type: object
            - type: array
              items:
                type: object
```

**Comment by arturdzm** [comment]
Currently the oneOf, anyOf, allOf scanning is limited to class level annotation and doesn't fully support java simple types, eg Object/List.

Currently only this scenario should work 
```
@Schema(oneOf = { Cat.class, Dog.class }
 public class Pet 
```

**Comment by Azquelt** [comment]
This issue is not present in `mpOpenApi-2.0` so if you're able to move up to that version, you won't hit this problem any more.

**Comment by srnjak** [comment]
I can confirm that issue is resolved in `mpOpenAPI-2.0` using OpenLiberty `21.0.0.3`.
